src/v/cluster/rm_stm.cc

```diff
@@ -2247,6 +2270,9 @@ ss::future<tx_errc> rm_stm::do_try_abort_old_tx(model::producer_identity pid) {
                       "applied pid:{} tx_seq:{}",
                       pid,
                       tx_seq);
+                    if (_c->is_leader() && _c->term() == synced_term) {
+                        co_await _c->step_down("try_abort(commit) apply error");
+                    }
```

Comment:
q: whats the rationale behind all these strategically placed step downs?  Maybe good to capture in the commit message.

Reply:
put it in a separate commit and add commit message